yunwoo/programmers-1/약수의-개수와-덧셈.js

```diff
@@ -0,0 +1,49 @@
+function solution(left, right) {
+  let result = 0;
+  const numArr = new Array(right - left + 1).fill().map((_, i) => left + i);
+
+  for (let element of numArr) {
+    let i = 1;
+    const divisorArr = [];
+
+    while (i <= element) {
+      if (element % i === 0) {
+        divisorArr.push(i);
+      }
+      i++;
+    }
+
+    !(divisorArr.length % 2) ? (result += element) : (result -= element);
+  }
+
+  return result;
+}
+
+function solution(left, right) {
+  let result = 0;
+
+  for (let i = left; i <= right; i++) {
+    const divisorArr = [];
+
+    for (let j = 1; j <= i; j++) {
+      if (i % j === 0) {
+        divisorArr.push(j);
+      }
+    }
+
+    !(divisorArr.length % 2) ? (result += i) : (result -= i);
+  }
+  return result;
+}
```

Comment:
```suggestion
function solution(left, right) {
  let result = 0;

  for (let i = left; i <= right; i++) {
    let divisorCount = 0;

    for (let j = 1; j <= i; j++) {
      if (i % j === 0) {
        divisorCount++;
      }
    }
    result += !(divisorCount % 2) ? i : -i;
  }
  
  return result;
}
```
- 아래 코드는 약수의 개수를 가져오는 `함수를 따로 분리`해서 작성한 코드입니다. 참고해주세요
```js
function solution(left, right) {
  let result = 0;

  for (let i = left; i <= right; i++) {
    result += getDivisorCount(i) % 2 === 0 ? i : -i;
  }

  return result;
}

function getDivisorCount(num) {
  let divisorCount = 0;

  for (let i = 1; i <= num; i++) {
    if (num % i === 0) divisorCount++;
  }

  return divisorCount;
}
```

Reply:
- 약수의 개수를 가져오는 역할의 함수를 분리하신 이유는 단일책임원칙이 맞을까요??🤔
- 저는 함수자체가 약수를 판별하고 값을 더해주는 한가지 일을 하는 함수라고 생각했는데 이렇게 세분화 된게 훨씬 좋은거 같습니다 잘봤습니다 감사합니다👍🙇‍♂️